### PR TITLE
Use cms menuitems to create footer on website #7561

### PR DESCRIPTION
**Pull request by aroraumang**

### CODE REVIEW

File: templates/webshop/base.jinja [hunk 1]
```diff
@@ -254,19 +252,28 @@
...
-          <div class="col-md-2">
-            {% set category = get_article_category(category_name) %}
-            <h4>{{ category.title }}</h4>
-            <ul class="list-unstyled">
-              {% for article in category.articles %}
-              <li><a href="{{ url_for('nereid.cms.article.render', uri=article.uri) }}">{{ article.title }}</a></li>
-              {% endfor %}
-            </ul>
-          </div>
+          {% set footer_menuitem = get_using_xml_id('nereid_webshop', 'webshop_root_cms_footer').get_menu_item(max_depth=10) %}
```
Comment (tarunbhardwaj) [inline templates/webshop/base.jinja]: `nereid_website.cms_root_footer`

File: templates/webshop/base.jinja [hunk 2]
```diff
@@ -254,19 +252,28 @@
...
+          {% for child in footer_menuitem.children %}
+            <div class="col-md-2">
+              <h4>{{ child.title }}</h4>
+              <ul class="list-unstyled">
+                {% for sub_child in child.children recursive %}
+                <li>
+                  {% if sub_child.children %}
+                    <br>
+                    <a href="{{ sub_child.link }}"><h5>{{ sub_child.title }}</h5></a>
+                    {{ loop(sub_child.children) }}
```
Comment (tarunbhardwaj) [inline templates/webshop/base.jinja]: not sure if footer can be hierarchical. IMO we should just consider 1st child

File: webshop.py [hunk 3]
```diff
@@ -113,6 +113,11 @@ class Website:
 
     copyright_year_range = fields.Char('Copyright Year Range')
 
+    cms_root_footer = fields.Many2One(
```
Comment (tarunbhardwaj) [inline webshop.py]: default can be the record in xml

File: cms.xml [hunk 4]
```diff
@@ -10,6 +10,14 @@
             <field name="target">_self</field>
         </record>
 
+        <!-- webshop root footer navigation -->
+        <record id="cms_root_footer" model="nereid.cms.menuitem">
```
Comment (prakashpp) [inline cms.xml]: also ensure if this record could be edited!

File: webshop.py (hunk 3 above)
Comment (aroraumang) [inline webshop.py]: Done

File: cms.xml (hunk 4 above)
Comment (aroraumang) [inline cms.xml]: Done

**Pull request merged**